python_modules/dagster/dagster/_core/storage/dagster_run.py

```diff
@@ -608,6 +612,22 @@ def __new__(
             end_time=check.opt_float_param(end_time, "end_time"),
         )
 
+    @classmethod
+    async def _batch_load(
+        cls,
+        keys: Iterable[str],
+        instance: "DagsterInstance",
+    ) -> Iterable[Optional["RunRecord"]]:
+        result_map: Dict[str, Optional[RunRecord]] = {run_id: None for run_id in keys}
+
+        # this should be replaced with an async DB call
+        records = instance.get_run_records(RunsFilter(run_ids=list(result_map.keys())))
+
+        for record in records:
+            result_map[record.dagster_run.run_id] = record
+
+        return result_map.values()
+
```

Comment:
hmmm should this interface return a dict? Seems less error-prone than a list that has to line up with the incoming keys.